test/node-api.js

```diff
@@ -6,11 +6,188 @@ chai.use(codeEquality);
 
 const { expect } = chai;
 
-const p = new Preprocessor();
+describe("parse", function() {
+  it("basic example", function() {
+    let p = new Preprocessor();
```

Comment:
Why does this switch to instantiated the Preprocessor per test?

(Not saying that's wrong, just curious if some bug was revealed when we were sharing them.)

Reply:
The `Preprocessor` is somehow stateful. The number from the spans do not reset between calls and keeps accumulating. Incidentally, https://astexplorer.net suffer from the same bug: if you choose swc, type some code and refresh, it will work the first time, but after words, every time you edit the code, the span indices keeps growing and the hover/highlighting breaks. I figured that is probably something we want to fix, but it seems unrelated enough to the PR and wasn't super trivial.

Reply:
Ah, well that is pretty important. This PR is the first time anybody is relying on the spans, and the public API assumes you will keep reusing a Preprocessor. So if that doesn't actually work, there's no point in exposing Preprocessor instances as the API. 

Originally, I did go digging through swc to try to see how they handle this in incremental rebuilds and I didn't find a satisfactory answer. The shared state is supposed to be for global string interning as well as accumulating source map information. https://github.com/embroider-build/content-tag/issues/12 is tracking the fact that I didn't really trust that what we're going (which I copied from how swc does it) isn't accumulating more and more state without bound. Your use of the spans proves that it definitely is, and in a bad way.

I think the safest thing for us to do is to simplify the bindings so we just expose functions, not methods, and each call can initialize the state it needs afresh.

Reply:
I think that’s certainly a fine option, though I personally like an object based API, in that I am planning on working on the swc fork to support other kind of tags. I suspect a stable preprocessor object would be useful for storing configuration (eg allowed types of tags and/or how to transform them). If that sounds useful, I don’t think it would be a problem to keep the JS side api the same while resetting the state on each call. We could just make the struct empty and internally set up the needed state inside the method until we figure out how we are supposed to use the swc api?

Reply:
Yeah that would be OK. It does make sense to hold future config.

Reply:
Fixed (in the binding layer), I added some TODO to investigate what can be safely reused (it's probably not 0%)